tools/uniform_sampling.cpp

```diff
@@ -93,20 +93,34 @@ compute (const pcl::PCLPointCloud2::ConstPtr &input, pcl::PCLPointCloud2 &output
   // Estimate
   TicToc tt;
   tt.tic ();
-  
+
   print_highlight (stderr, "Computing ");
 
-  UniformSampling<PointXYZ> us;
-  us.setInputCloud (xyz);
-  us.setRadiusSearch (radius);
-  PointCloud<PointXYZ> output_;
-  us.filter (output_);
+  UniformSampling<PointXYZ> us(true); // extract removed indices
+  us.setInputCloud(xyz);
+  us.setRadiusSearch(radius);
+  PointCloud<PointXYZ> temp;
+  us.filter(temp);
+
+  pcl::PointIndices removed_indices;
+  // These are "fake" indices generated by initCompute(), already sorted
+  auto input_indices = us.getIndices();
+  // Removed indices, not guaranteed to be sorted
+  us.getRemovedIndices(removed_indices);
+  std::sort(removed_indices.indices.begin(),
+            removed_indices.indices.end(),
+            [](const auto& a, const auto& b) { return a < b; });
```

Comment:
`removed_indices.indices` is `std::vector<int>` and the comparison metric you're using is the default one. I suspect you don't need to provide this third argument.